src/SFA.DAS.FAA.Web/Controllers/SavedSearchesController.cs

```diff
@@ -0,0 +1,74 @@
+﻿using MediatR;
+using Microsoft.AspNetCore.Mvc;
+using SFA.DAS.FAA.Web.Infrastructure;
+using SFA.DAS.FAA.Web.Models.SavedSearches;
+using SFA.DAS.FAA.Application.Commands.SavedSearches.DeleteSavedSearch;
+using SFA.DAS.FAA.Application.Queries.SavedSearches.GetConfirmUnsubscribe;
+using SFA.DAS.FAA.Web.Models.User;
+
+namespace SFA.DAS.FAA.Web.Controllers
+{
+    
+    [Route("saved-searches")]
+    public class SavedSearchesController(IMediator mediator, IDataProtectorService dataProtectorService, ICacheStorageService cacheStorageService) : Controller
+    {
+        [HttpGet]
+        [Route("unsubscribe", Name = RouteNames.GetSavedSearchesUnsubscribe)]
+        public async Task<IActionResult> Index([FromQuery]string id)
+        {            
+            var savedSearchId = dataProtectorService.DecodeData(id);
+            if (!Guid.TryParse(savedSearchId, out var searchId)) 
+            {
+                return RedirectToRoute(RouteNames.ServiceStartDefault);
+            }
+            
+            var result = await mediator.Send(new GetConfirmUnsubscribeQuery
+            {
+                SavedSearchId = searchId
+            });
+
+            if(result.SavedSearch == null)
+            {
+                return RedirectToRoute(RouteNames.ServiceStartDefault);
+            }
+
+            var viewModel = new UnsubscribeSavedSearchesViewModel
+            {
+                SavedSearch = SavedSearchViewModel.From(result.SavedSearch, result.Routes)
+            };
+
+            return View(viewModel);
+        }
+
+        [HttpPost]
+        [Route("unsubscribe", Name = RouteNames.PostSavedSearchesUnsubscribe)]
+        public async Task<IActionResult> Index(UnsubscribeSavedSearchesModel model)
+        {
+            await mediator.Send(new UnsubscribeSavedSearchCommand
+            {
+                SavedSearchId = model.Id
+            });
+
+            await cacheStorageService.Set($"{model.Id}-savedsearch", model, 5, 5);
```

Comment:
Feels like there could be an injection into the cache of a large amount of data here?